File: php/spec/reporters/html.spec.php
<?php declare(strict_types=1);

use Eno\Reporters\HTML;

describe('HTML reporter', function() {
  given('_context', function() {
    require(__DIR__ . '/context_fixture.php');
    return $context;
  });

  it('produces text output', function() {
    $this->_context->reporter = new HTML;

    $snippet = HTML::report(
      $this->_context,
      $this->_context->instructions[1],
      $this->_context->instructions[0]
    );

    expect($snippet)->toMatchSnapshot('spec/reporters/snapshots/html.snap.html');
  });
});

File: ruby/spec/reporters/html.spec.rb
# frozen_string_literal: true

input = <<~DOC.strip
> comment
# section

field: value

field_with_items:
- item
- item

> comment
- item

## subsection

field_with_attributes:
attribute = value

> comment
attribute = value
DOC

describe Enolib::HtmlReporter do
  it 'produces html output' do
    document = Enolib.parse(input, reporter: Enolib::HtmlReporter)
    context = document.instance_variable_get(:@context)

    reporter = context.reporter.new(context)
    snippet = reporter.report_element(context.document[:elements].first).snippet()

    expect(snippet).to match_snapshot
  end
end
